src/base/tests/test_util.py

```diff
@@ -485,6 +485,27 @@ def test_explode_mult_filters(self):
             cr.execute(q)
             self.assertTrue(all(x for (x,) in cr.fetchall()))
 
+    def test_explode_query_range(self):
+        cr = self.env.cr
+
+        cr.execute("SELECT count(id) FROM res_partner_title")
+        count = cr.fetchone()[0]
+        # ensure there start with at least 10 records
+        for _ in range(10 - count):
+            count += 1
+            self.env["res.partner.title"].create({"name": "x"})
+
+        # set one record with very high id
+        tid = self.env["res.partner.title"].create({"name": "x"}).id
+        count += 1
+        cr.execute("UPDATE res_partner_title SET id = 10000000 WHERE id = %s", [tid])
+
+        qs = util.explode_query_range(cr, "SELECT 1", table="res_partner_title", bucket_size=count)
+        self.assertEqual(len(qs), 1)  # one bucket should be enough for all records
+
+        qs = util.explode_query_range(cr, "SELECT 1", table="res_partner_title", bucket_size=count - 1)
+        self.assertEqual(len(qs), 1)  # 10% rule for second bucket, 1 <= 0.1(count - 1) since count >= 11
```

Comment:
Only true if count is 11.
This is not always the case, like with `l10n_cr`.

Reply:
This is true if count >**=** 11, which is guaranteed by
https://github.com/odoo/upgrade-util/pull/56/files#diff-d21f53fbfb10b7592abbfa97c35cfb969bbf545b9e36a44ed5069b59f438e350R493-R499

Reply:
In other words the record created in line 499 is at _least_ the 11th.